hal/list.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Operating system kernel
+ *
+ * List
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Hubert Badocha
```

Comment:
This implementation was adapted from elsewhere, would be nice to mention the OG too